backend/base/views/user_views.py

```diff
@@ -33,105 +38,115 @@ class MyTokenObtainPairView(TokenObtainPairView):
-    return Response(serializer.data)  # now the data comes from the database
+        if user.email != "":
+            user.username = user.email
 
+        user.save()
 
-# if token is off, then access is denied because authentication
-# credentials were not provided
-@api_view(["GET"])
-@permission_classes([IsAuthenticated])  # protected route
-def getUserProfile(request):
-    user = request.user
-    serializer = UserSerializer(
-        user, many=False
-    )  # 'many=False' means that I want to obtain one user
-    # return Response(user) <- it's incorrect because it's NOT serialized
-    return Response(serializer.data)  # now the data comes from the database
+        # return Response(user) <- it's incorrect because it's NOT serialized
+        return Response(serializer.data)
 
+    def get(self, request, format=None):
+        # get user profile
+        user = request.user
+        serializer = UserSerializer(user, many=False)
+        # return Response(user) <- it's incorrect because it's NOT serialized
+        return Response(serializer.data)
 
-@api_view(["GET"])
-@permission_classes([IsAdminUser])  # only for admins
-def getUsers(request):
-    users = (
-        User.objects.all()
-    )  # it takes data from the database, because User is a model
-    serializer = UserSerializer(
-        users, many=True
-    )  # 'many=True' means that I'm passing multiple objects
-    # return Response(users) <- it's incorrect because
-    # the User object is NOT serialized
-    return Response(serializer.data)  # now the data comes from the database
 
+class AdminProfile(APIView):
+    """
+    User management for admins
+    """
 
-@api_view(["GET"])
-@permission_classes([IsAdminUser])
-def getUserById(request, pk):
-    user = User.objects.get(id=pk)
-    serializer = UserSerializer(user, many=False)
-    return Response(serializer.data)
+    permission_classes = [IsAdminUser]
 
+    def get(self, request, format=None):
+        # get users
+        users = User.objects.all()
+        serializer = UserSerializer(users, many=True)
+        # return Response(users) <- it's incorrect because
+        # the User object is NOT serialized
+        return Response(serializer.data)
 
-@api_view(["PUT"])
-@permission_classes([IsAuthenticated])
-def updateUser(request, pk):
-    user = User.objects.get(id=pk)
+    def put(self, request, pk, format=None):
+        # update user
+        user = get_object_or_404(User, id=pk)
+        data = request.data
 
-    data = request.data
+        serializer = UserUpdateSerializer(
+            user,
+            data=data,
+            many=False,
+        )
+        serializer.is_valid()
+        # print(serializer.errors)
 
-    user.first_name = data["name"]
-    user.username = data["email"]
-    user.email = data["email"]
-    user.is_staff = data["isAdmin"]
+        serializer.save(first_name=data["name"], is_staff=data["isAdmin"])
```

Comment:
to lepiej przemapować przed linią 122, żeby było poprawnie zwalidowane i zapisane automatycznie przez serializer

Reply:
done